### PR TITLE
fs: shell: littlefs: add support of block devices

### PR DESCRIPTION
**Pull request by xakep-amatop**
For flash device, littlefs should be located at the partition. Current implementation of the fs shell is hardcoded for the flash device and the compilation fails if STORAGE_PARTITION isn't defined in the device tree.
Add options for block dev support. This allows to mount littlefs on the block device from the shell, and also removes the compile time dependency on STORAGE_PARTITION, if CONFIG_FS_LITTLEFS_BLK_DEV is set.

This is an incremental improvement, ideally, someone should implement selection of the storage dev from the shell args.

### CODE REVIEW

**Comment by xakep-amatop** [comment]
Hello, @Laczen @de-nordic
I would really appreciate it if you could review this PR.

File: tests/subsys/fs/littlefs/src/testfs_lfs.c [hunk 1]
```diff
@@ -27,7 +27,7 @@ struct fs_mount_t testfs_small_mnt = {
 };
 
 #if CONFIG_APP_TEST_CUSTOM
-FS_LITTLEFS_DECLARE_CUSTOM_CONFIG(medium, MEDIUM_IO_SIZE, MEDIUM_IO_SIZE,
```
Comment (de-nordic) [inline tests/subsys/fs/littlefs/src/testfs_lfs.c:30]: Shouldn't this be rather size of pointer or int? Do we have some define that shows what is platform arch/platform alignment?

File: include/zephyr/fs/littlefs.h [hunk 2]
```diff
@@ -62,15 +62,17 @@ struct fs_littlefs {
...
  * @param read_sz see @kconfig{CONFIG_FS_LITTLEFS_READ_SIZE}
  * @param prog_sz see @kconfig{CONFIG_FS_LITTLEFS_PROG_SIZE}
  * @param cache_sz see @kconfig{CONFIG_FS_LITTLEFS_CACHE_SIZE}
  * @param lookahead_sz see @kconfig{CONFIG_FS_LITTLEFS_LOOKAHEAD_SIZE}
  */
-#define FS_LITTLEFS_DECLARE_CUSTOM_CONFIG(name, read_sz, prog_sz, cache_sz, lookahead_sz) \
-	static uint8_t __aligned(4) name ## _read_buffer[cache_sz];			  \
-	static uint8_t __aligned(4) name ## _prog_buffer[cache_sz];			  \
-	static uint32_t name ## _lookahead_buffer[(lookahead_sz) / sizeof(uint32_t)];		  \
+#define FS_LITTLEFS_DECLARE_CUSTOM_CONFIG(name, alignment, read_sz, prog_sz, cache_sz,	  \
```
Comment (de-nordic) [inline include/zephyr/fs/littlefs.h:71]: Change should be mentioned in release notes.

Comment (xakep-amatop) [reply include/zephyr/fs/littlefs.h:71]: done

**Review by de-nordic — APPROVED** [review]

**Review by Laczen — APPROVED** [review]

**Pull request merged**